python/ray/data/tests/test_util.py

```diff
@@ -19,6 +20,20 @@
 from ray.data.tests.conftest import *  # noqa: F401, F403
 
 
+def test_cached_remote_fn():
+    def foo():
+        pass
+
+    cpu_only_foo = cached_remote_fn(foo, num_cpus=1)
+    cached_cpu_only_foo = cached_remote_fn(foo, num_cpus=1)
+
+    assert cpu_only_foo == cached_cpu_only_foo
+
+    gpu_only_foo = cached_remote_fn(foo, num_gpus=1)
+
+    assert cpu_only_foo != gpu_only_foo
```

Comment:
can you also add an E2E test to check that the tasks are actually running in different workers?

Reply:
Yeah, makes sense